cockpit/devices/boulderSLM.py

```diff
@@ -44,12 +44,45 @@
 
 from cockpit import events
 import cockpit.gui.device
-import cockpit.gui.guiUtils
-import cockpit.gui.toggleButton
 import cockpit.handlers.executor
 import time
 import cockpit.util
 
+class LastParameters(object):
+    """A class to keep a record of last SIM parmeters using async calls."""
+    def __init__(self, slm):
```

Comment:
Yeah - I keep hitting this (anti)pattern.
The LastParameters class is responsible for handling an asynchronous call to the remote device to update a record of the last-used parameters. It needs either a reference to the device or to its asynchronous proxy to do this. I thought it made more sense to pass the reference to the device once, rather than every time this is called, because:
* the device has a singleton instance of LastParameters;
* that singleton always calls acts on the same device.
I guess the update method could go on the device and act on the LastParameters instance ... but then you'd have an outer class updating a private member of an inner class.

Reply:
Any more ideas on this one?

Reply:
New issue #473